src/Rector/FuncCall/RemoveDumpDataDeadCodeRector.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Laravel\Rector\FuncCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\StaticCall;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Laravel\Tests\Rector\FuncCall\RemoveDumpDataDeadCodeRector\RemoveDumpDataDeadCodeRectorTest
+ */
+
+final class RemoveDumpDataDeadCodeRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'It will removes the dump data just like dd or dump functions from the code.`',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class MyController
+{
+    public function store()
+    {
+        dd('test');
+        return true;
+    }
+
+    public function update()
+    {
+        dump('test');
+        return true;
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class MyController
+{
+    public function store()
+    {
+        return true;
+    }
+
+    public function update()
+    {
+        return true;
+    }
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class];
```

Comment:
It should use Expression instead, so check against Expression expr is FuncCall direct remove FuncCall on If_ cond possibly cause error

Reply:
Another approach is using FuncCall, but when remove, verify first that parent attribute is Expression stmt